addon/-private/system/references/belongs-to.js

```diff
@@ -63,32 +63,33 @@ BelongsToReference.prototype.push = function(objectOrPromise) {
   });
 };
 
-BelongsToReference.prototype.value = function() {
-  var inverseRecord = this.belongsToRelationship.inverseRecord;
+BelongsToReference.prototype.isLoaded = function() {
+  let inverseRecord = this.belongsToRelationship.inverseRecord;
+
+  return inverseRecord && inverseRecord.isLoaded();
+};
 
-  if (inverseRecord && inverseRecord.record) {
-    return inverseRecord.record;
+BelongsToReference.prototype.value = function() {
+  if (this.isLoaded()) {
+    return this.belongsToRelationship.inverseRecord.getRecord();
   }
 
   return null;
 };
 
 BelongsToReference.prototype.load = function() {
-  if (this.remoteType() === "id") {
-    return this.belongsToRelationship.getRecord();
-  }
+  let record = this.value();
```

Comment:
Actually this seems weird. So first you do `this.value()` which is  `this.belongsToRelationship.inverseRecord.getRecord()`  but then if it's null you do `this.belongsToRelationship.getRecord();` Do we just have two different things named `getRecord`? One on intneralModel and the other on the relationship? Maybe we should rename?